framework/metricsresource/resource.go

```diff
@@ -0,0 +1,188 @@
+package metricsresource
```

Comment:
My main goal is to add a metrics resource. Some changes to the `Resource` interface which I try to explain in other comments have been necessary.